app/Policies/CoursePolicy.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Policies;
+
+use App\Model\Course;
+use App\Model\Instructor;
+
+class CoursePolicy
+{
+    public function update(Instructor $instructor, Course $course): bool
+    {
+        // 管理者以外の場合
+        if ($instructor->id === $course->instructor_id) {
+            return true;
+        }
+
+        // 管理者の場合
+        if ($instructor->isManager()) {
+            $manager = Instructor::with('managings')->find($instructor->id);
+
+            if (! $manager) {
+                return false;
+            }
+
+            $managerIds = $manager->managings->pluck('id')->toArray();
+
+            return in_array($course->instructor_id, $managerIds);
+        }
+
+        return false;
```

Comment:
コードを最適化して、次のようにお願いできますか？
このコードにおいて、不明点があればお聞きください！

```suggestion
        if ($instructor->isManager()) {
            // 管理者の場合、配下の講師の講座も更新可能
            $managerIds = $instructor->managings->pluck('id')->toArray();
            $managerIds[] = $instructor->id;

            return in_array($course->instructor_id, $managerIds, true);
        }


        // 講師の場合、自分の講座のみ更新可能
        return $instructor->id === $course->instructor_id;
```